src/pudl/transform/eia861.py

```diff
@@ -1020,6 +1014,86 @@ def _harvest_associations(dfs: list[pd.DataFrame], cols: list[str]) -> pd.DataFr
     return assn
 
 
+def _combine_88888_values(df: pd.DataFrame, idx_cols: list[str]) -> pd.DataFrame:
+    """Combine 88888 utility_id_values.
+
+    This function sums rows with a utility_id_eia of 88888 into a single row by
+    primary key. It drops rows with a utility_id_eia of 88888 if there are non-numeric
+    columns with different values that are impossible to combine. E.g.: boolean
+    columns where one value is Y and the other is N. This results in a small loss of
+    data, but is necessary to ensure that the primary key remains unique.
+
+    There's a similar de-duplication process happening in the ``_tidy_class_dfs`` function,
+    but the procedure for 88888 has been abstracted out here to avoid confusion and the
+    ability to treat these cases differently as we see fit. The 88888 utility_id_eia
+    values are known withholdings whereas the duplicate primary key rows are unknown data
+    quality issues.
+
+    Args:
+        df: The dataframe to process.
+        idx_cols: The index (primary key) columns of the input dataframe. They must
+            identify unique records in the input data.
+
+    Returns:
+        A dataframe with the combined or dropped 88888 utiltiy_id_eia values and unique pks.
+    """
+    if 88888 not in df.utility_id_eia.unique():
+        return df
+    non_num_cols = (
+        df.set_index(idx_cols).select_dtypes(exclude="number").columns.tolist()
+    )
+
+    def custom_group_agg(group):
```

Comment:
nit: `custom_group_agg` doesn't really tell me much about what this function does... something like `sum_only_numeric_values` might be a little clearer